etc/config/csharp.amazon.properties

```diff
@@ -23,6 +23,6 @@ compiler.dotnet700csharp.langVersion=latest
 compiler.dotnettrunkcsharp.exe=/opt/compiler-explorer/dotnet-trunk/.dotnet/dotnet
 compiler.dotnettrunkcsharp.name=.NET (main)
 compiler.dotnettrunkcsharp.clrDir=/opt/compiler-explorer/dotnet-trunk
-compiler.dotnettrunkcsharp.targetFramework=net8.0
+compiler.dotnettrunkcsharp.targetFramework=net7.0
```

Comment:
@mattgodbolt, is it possible to dynamically set this property by reading https://github.com/dotnet/runtime/blob/main/global.json#L3? global.json's format hasn't changed since .net core v1, so reading first three characters of `sdk.version` (`7` `.` `0`) and prepending `net` is reliable and it will save us from updating `main` manually.

Reply:
We don't have global.json in Core_Root so it's not feasible. 
A possible way is to detect the prefix of the directory name in Core_Root/.dotnet/shared/Microsoft.NETCore.App to see which version of libs we have and set the corresponding TFM.

Reply:
if downloading a file is not possible here, then we can copy global.json to archive dir: `cp global.json ${CORE_ROOT}` after https://github.com/compiler-explorer/misc-builder/blob/3721571a5fcb45f3d6caf167f25832203b8ab5f9/build/build-dotnet.sh#L58 and extract the value like this:

```js
// a.mjs (run as `node a.mjs` with latest node: v19)
import data from './global.json' assert { type: 'json' };

const parts = data.sdk.version.split('.');
const targetFramework = `net${parts[0]}.${parts[1]}`;
```

i just don't know how to assign these properties dynamically

Reply:
if downloading a file is not possible here, then we can copy global.json to archive dir: `cp global.json ${CORE_ROOT}` after https://github.com/compiler-explorer/misc-builder/blob/3721571a5fcb45f3d6caf167f25832203b8ab5f9/build/build-dotnet.sh#L58 and extract the value like this:

```js
// a.mjs (run as `node a.mjs` with latest node: v19)
import data from './global.json' assert { type: 'json' };

const parts = data.sdk.version.split('.');
const targetFramework = `net${parts[0]}.${parts[1]}`;
```

i just don't know how to dynamically assign these properties

Reply:
nvm, figured it out, it's in dotnet.ts

https://github.com/compiler-explorer/misc-builder/pull/54

i'll update dotnet.ts once that is merged.

Reply:
> nvm, figured it out, it's in dotnet.ts
> 
> https://github.com/compiler-explorer/misc-builder/pull/54
> 
> i'll update dotnet.ts once that is merged.

Including a global.json will also result in dependencies in global.json being restored which is not expected (we don't have Internet in the production environment), and will cause failure to restore. 

You can reuse the SDK version here: https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/compilers/dotnet.ts#L125-L125

Reply:
```typescript
const versionSegments = sdkVersions[0].split('.');
const targetFramwok = `${versionSegments[0]}.{versionSegments[1]}`;
```

I think this should be enough. And in this way you can also remove the `targetFramwork` config in properties files.